js/apps/filter-page/services/filterService.js

```diff
@@ -0,0 +1,73 @@
+((app) => {
+	const filterService = () => {
+		const getAllTagInfo = (dataWithTags) => {
+			let tags = [];
+
+			angular.forEach(dataWithTags, (dataItem) => {
+				tags = tags.concat(dataItem.Tags);
+			});
+
+			return tags;
+		};
+
+		const getFamilies = (tagInfoArr) => {
+			return _.uniq(_.pluck(tagInfoArr, 'Name'), name => name);
+		};
+
+		const getFamilyTags = (familyName, tagInfoArr) => {
+			const familyTagInfo = _.where(tagInfoArr, { Name: familyName });
+			return _.uniq(_.sortBy(_.pluck(familyTagInfo, 'Tag'), tag => tag), tag => tag);
+		};
+
+		const getFamilyType = (familyName, tagInfoArr) => {
+			return _.findWhere(tagInfoArr, { Name: familyName }).Type;
```

Comment:
If there are not results, will this throw an error?

Reply:
yep - done